metrics/api/src/main/java/io/helidon/metrics/api/MetricsConfigBlueprint.java

```diff
@@ -0,0 +1,273 @@
+ *         <p>
+ *             Specifically, users can specify whether scope tags are used at all and, if so, what tag name to use.
+ *         </li>
+ *     </ul>
+ */
+@ConfigBean()
+@Configured(root = true, prefix = MetricsConfigBlueprint.METRICS_CONFIG_KEY)
+@Prototype.Blueprint(decorator = MetricsConfigBlueprint.BuilderDecorator.class)
+@Prototype.CustomMethods(MetricsConfigSupport.class)
+interface MetricsConfigBlueprint {
+
+    /**
+     * The config key containing settings for all of metrics.
+     */
+    String METRICS_CONFIG_KEY = "metrics";
+
+    /**
+     * Config key for comma-separated, {@code tag=value} global tag settings.
+     */
+    String GLOBAL_TAGS_CONFIG_KEY = "tags";
+
+    /**
+     * Config key for the app tag value to be applied to all metrics in this application.
+     */
+    String APP_TAG_CONFIG_KEY = "app-name";
+
+    /**
+     * Config key for scope-related settings.
+     */
+    String SCOPE_CONFIG_KEY = "scope";
+
+    /**
+     * Config key for the default scope value.
+     */
+    String SCOPE_DEFAULT_VALUE_CONFIG_KEY = "default";
+
+    /**
+     * Config key for tag settings within the scope config section.
+     */
+    String SCOPE_TAG_CONFIG_KEY = "tag";
+
+    /**
+     * Config key for the tag name to use for storing scope values within the scope config section.
+     */
+    String SCOPE_TAG_NAME_CONFIG_KEY = "name";
+
+    /**
+     * Default tag name to use for recording a meter's scope as a tag (if that behavior is enabled).
+     */
+    String SCOPE_TAG_NAME_DEFAULT = "scope";
+
+    /**
+     * Config key for whether to store scopes as tags.
+     */
+    String SCOPE_TAG_ENABLED_CONFIG_KEY = "enabled";
+
+    /**
+     * Default setting for whether to use scope tags (as text).
+     */
+    String SCOPE_TAG_ENABLED_DEFAULT_VALUE = "false";
+
+    /**
+     * Default setting for whether to use scope tags (as boolean).
+     */
+    boolean SCOPE_TAG_ENABLED_DEFAULT = Boolean.parseBoolean(SCOPE_TAG_ENABLED_DEFAULT_VALUE);
+
+
+    /**
+     * Whether metrics functionality is enabled.
+     *
+     * @return if metrics are configured to be enabled
+     */
+    @ConfiguredOption("true")
+    boolean enabled();
+
+    /**
+     * Key performance indicator metrics settings.
+     *
+     * @return key performance indicator metrics settings
+     */
+    @ConfiguredOption(key = KeyPerformanceIndicatorMetricsConfigBlueprint.KEY_PERFORMANCE_INDICATORS_CONFIG_KEY)
+    KeyPerformanceIndicatorMetricsConfig keyPerformanceIndicatorMetricsConfig();
+
+    /**
+     * Global tags.
+     *
+     * @return name/value pairs for global tags
+     */
+    @ConfiguredOption(key = GLOBAL_TAGS_CONFIG_KEY)
+    List<Tag> globalTags();
+
+    /**
+     * Application tag value added to each meter ID.
+     *
+     * @return application tag value
+     */
+    @ConfiguredOption(key = APP_TAG_CONFIG_KEY)
```

Comment:
The `APP_TAG_CONFIG_KEY` seems to be `app-name`, so the method should be called `appName`, or the key should be `app-tag-value`.
I prefer consistency (same key as method name), as then we do not need to define the key explicitly and it is more readable.

Reply:
changed